common/junit-platform-native/src/main/java/org/graalvm/junit/platform/JUnitPlatformFeature.java

```diff
@@ -156,20 +157,31 @@ private TestPlan registerTestPlan(Launcher launcher, List<Path> classpath) {
                 .map(Optional::get)
                 .filter(ClassSource.class::isInstance)
                 .map(ClassSource.class::cast)
-                .forEach((ClassSource classSource) -> {
-                    Class<?> clazz = classSource.getJavaClass();
-                    if (debug) {
-                        System.out.println("[Debug] Found test class: " + clazz);
-                    }
-                    RuntimeReflection.registerForReflectiveInstantiation(clazz);
-                    for (Field field : clazz.getDeclaredFields()) {
-                        RuntimeReflection.register(field);
-                    }
-                    for (Method method : clazz.getDeclaredMethods()) {
-                        RuntimeReflection.register(method);
-                    }
-                });
+                .forEach(this::registerTestClassForReflection);
 
         return testPlan;
     }
+
+    private void registerTestClassForReflection(ClassSource classSource) {
+        Class<?> clazz = classSource.getJavaClass();
+        if (debug) {
+            System.out.println("[Debug] Found test class: " + clazz);
+        }
+
+        Class<?> declaringClass = clazz.getDeclaringClass();
```

Comment:
We could also do `clazz.getEnclosingClass` here - the only difference between `getDeclaringClass` and `getEnclosingClass` is that `getEnclosingClass` also works for anonymous classes. However, I don't think that's relevant in this context, but I may be wrong

Reply:
JUnit 4 and JUnit Jupiter do not support anonymous classes as test classes, but that doesn't mean that some other `TestEngine` in the world doesn't.

I think it might be unlikely that there's such a `TestEngine`, but I can't 100% rule it out.

So maybe going on the safe side (using `getEnclosingClass()`) is the better choice, especially if it has no adverse side effects.

Reply:
Ah, this is a good point - I'll change it to `getEnclosingClass`